app/cdap/components/PipelineDiff/util/diff.ts

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { diff } from 'json-diff';
+import { DeepPartial, DiffIndicator, IPipelineDiffMap } from '../types';
+import {
+  IPipelineConfig,
+  IStageMap,
+  IConnectionMap,
+  IPipelineStage,
+  IPipelineConnection,
+} from '../types';
+
+/**
+ * @param stage stage
+ * @returns the key of the stage in the diff map
+ */
+export function getStageDiffKey(stage: IPipelineStage) {
+  return `${stage.plugin.name}\u0001${stage.name}\u0002`;
```

Comment:
Add some more comments why we want to construct keys using unicode

Reply:
Done.